explore/src/main/scala/explore/tabs/AsterismEditorTile.scala

```diff
@@ -44,12 +46,13 @@ object AsterismEditorTile:
     obsIds:            ObsIdSet,
     asterismIds:       View[AsterismIds],
     allTargets:        UndoSetter[TargetList],
+    programSummaries:  UndoContext[ProgramSummaries],
```

Comment:
I'd rather we limit write access to only what the lower components are supposed to change. Unless we need the lower components to be able modify any part of `ProgramSummaries`.

Also, passing `UndoContext[ProgramSummaries]` makes the `allTargets` property redundant.

Finally, `UndoContext` is an `UndoSetter` that adds the ability to `undo` and `redo`. If we don't plan to allow those in the lower components, it's enough to pass an `UndoSetter`.

Reply:
Lower down there is an `Action` that modifies both the observations and the targets in the ProgramSummaries. I guess I could make a `Lens` into a tuple of `(ObservationList, TargetList)` and use that to zoom into the ProgramSummaries. 

It does have to be an undo context, because the `Action` requires `UndoContext`

Reply:
Yes, we can do a `disjointZip` to zoom into observations and targets if we need them together. Then a `zoom` into the first element of the tuple would give us `allTargets`.

> the Action requires UndoContext

Not sure I follow what you meant. The `set` and `mod` methods in `Action` take an `UndoSetter`.

Reply:
> Not sure I follow what you meant. The `set` and `mod` methods in `Action` take an `UndoSetter`.

My bad. I just glanced at the class and the parameters are named `undoCtx`, but they are `UndoSetter`s.